cpp/include/cudf/aggregation.hpp

```diff
@@ -72,16 +72,32 @@ std::unique_ptr<aggregation> make_quantile_aggregation(
 */
 std::unique_ptr<aggregation> make_argmax_aggregation();
 
-
 /**
  * @brief Factory to create an `argmin` aggregation
  * 
  * `argmin` returns the index of the minimum element.
 */
 std::unique_ptr<aggregation> make_argmin_aggregation();
 
+/**
+ * @brief Factory to create a PTX aggregation
+ *
+ * @param[in] user_defined_aggregator A string which has the required aggregation
+ * @param[in] output_type expected output type
+ *
+ * @return aggregation unique pointer housing user_defined_aggregator PTX string.
+ */
+std::unique_ptr<aggregation> make_ptx_aggregation(std::string user_defined_aggregator, data_type output_type);
```

Comment:
`make_ptx_aggregation` and `make_cuda_aggregation` strike me as a little odd namingwise. It isn't a PTX aggregation, it's really a user-defined aggregation specified by a PTX string.

I think I still like the original idea of having a single factory called `make_udf_aggregation` that has a flag to select between CUDA/PTX and a `std::string const&` parameter for the code.